cmd/snapshotgc/snapshotgc.go

```diff
@@ -122,3 +123,46 @@ func getUnassociatedNSSnapshots(
 
 	return unAssociatedSnaps, nil
 }
+
+// Keep a certain amount of pr/non-pr snapshots
+func getSnapshotsForRemoval(
+	cl client.Client,
+	snapshots []applicationapiv1alpha1.Snapshot,
+	prSnapshotsToKeep int,
+	nonPrSnapshotsToKeep int,
+	logger logr.Logger,
+) []applicationapiv1alpha1.Snapshot {
+	sort.Slice(snapshots, func(i, j int) bool {
+		return snapshots[j].CreationTimestamp.Before(&snapshots[i].CreationTimestamp)
+	})
+
+	shortList := []applicationapiv1alpha1.Snapshot{}
+	keptPrSnaps := 0
+	keptNonPrSnaps := 0
+
+	for _, snap := range snapshots {
+		label, found := snap.GetLabels()["pac.test.appstudio.openshift.io/event-type"]
+		if found && label == "pull_request" && keptPrSnaps < prSnapshotsToKeep {
+			// if pr, and we did not keep enough PR snapshots -> discard from candidates
+			logger.V(1).Info(
+				"Discarding PR candidate",
+				"snapshot-name", snap.Name,
+				"pr-snapshot-kept", keptPrSnaps+1,
+			)
+			keptPrSnaps++
+			continue
+		} else if (!found || label != "pull_request") && keptNonPrSnaps < nonPrSnapshotsToKeep {
+			// same for non-pr
+			logger.V(1).Info(
+				"Discarding non-PR candidate",
```

Comment:
same here

Reply:
`non-PR` could be probably named like `push` snapshot

Reply:
> non-PR could be probably named like push snapshot

There was a discussion around this during refinement. The conclusion was that we should differentiate between PR snapshots and any other type of snapshot, as to not limit us to future possible snapshot types.

https://docs.google.com/document/d/105ojKC-G_NuDYMXNLauRZ4cgDIiUjIRVQCE1dbYwXNo/edit?disco=AAABBHWg93g